src/main/java/uk/gov/ons/ctp/common/event/EventPublisher.java

```diff
@@ -313,8 +378,28 @@ public String sendEvent(
           .with("routingKey", routingKey)
           .with("exception", e)
           .error("Failed to send event");
-      throw new EventPublishException(e);
+
+      if (!eventPersistence.isFirestorePersistenceSupported()) {
```

Comment:
I considered doing this at the time. Looking at it again the call is much tighter than I thought. As I see it the pros for the current implementation are:

- Allows an error to be logged which is expli

I don't think that the change would get rid of isFirestorePersistenceSupported() because we do want to be able to fail the first time an application thinks it's sending an event with persistence enabled but it's not active. We don't want to discover that discrepancy only at the point at which Rabbit has actually failed.

Reply:
The is..() method is getting deleted.
EventPersistance is becoming an optional.
Exception text needs to not mention Firestore